src/main/java/org/wildfly/security/auth/realm/ldap/X509EvidenceVerifier.java

```diff
@@ -124,7 +125,11 @@ public boolean verifyCertificate(X509Certificate certificate, Attributes attribu
             Attribute attribute = attributes.get(ldapAttribute);
             final int size = attribute.size();
             for (int i = 0; i < size; i++) {
-                if (certificate.getSubjectDN().getName().equals(attribute.get(i))) {
+
+                X500Principal certSubjectX500Principal = certificate.getSubjectX500Principal();
+                X500Principal attSubjectX500Principal = new X500Principal((String)attribute.get(i));
```

Comment:
attribute.get(i) can potentially return null, passing null into the X500Principal constructor would lead to a NullPointerException so I suggest that is checked first.

Also as attribute.get(i) returns an Object we should probably double check the type before casting it.